### PR TITLE
Implement -delete (#13)

### PR DESCRIPTION
**Pull request by Arcterus**
While -delete may be supported now, there are (as of yet) no tests.  I am planning to push another commit with tests, but if you want to merge it now that is fine as well (I am going to write tests anyway unless someone else does).

### CODE REVIEW

**Comment by mcharsley** [comment]
Looks good so far. Would prefer tests to be added before committing the changes.

In the meantime, I'll liase with Google's open source folks about the CLA/Googlebot thing. I'd assume we can drop that now uutils has taken the project over, but I'm not a lawyer...

**Comment by Arcterus** [comment]
Yeah, that's not a problem.  I'll probably add them tomorrow as it's pretty late here.

Alright, keep me posted about that. :)

**Comment by mcharsley** [comment]
The CLA webhook has now been removed :-)

File: src/find/matchers/delete.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        Box::new(DeleteMatcher::new())
+    }
+
+    fn delete(&self, file_path: &Path, file_type: FileType) -> io::Result<()> {
+        if file_type.is_file() || file_type.is_symlink() {
+            fs::remove_file(file_path)
+        } else if file_type.is_dir() {
+            fs::remove_dir(file_path)
+        } else {
+            unimplemented!() // TODO: not sure what find does for block devices, etc.
```
Comment (tavianator) [inline src/find/matchers/delete.rs]: GNU find just calls `unlink()` (or `unlinkat()`) in this case, which will handle every file type except directories.  Looks like that's what `remove_file()` [does too](https://doc.rust-lang.org/std/fs/fn.remove_file.html#platform-specific-behavior), so presumably

```rust
if file_type.is_dir() {
    fs::remove_dir(file_path);
} else {
    fs::remove_file(file_path);
}
```

is fine

File: src/find/matchers/delete.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+}
+
+impl Matcher for DeleteMatcher {
+    fn matches(&self, file_info: &DirEntry, _: &mut MatcherIO) -> bool {
+        match self.delete(file_info.path(), file_info.file_type()) {
```
Comment (tavianator) [inline src/find/matchers/delete.rs]: GNU find skips deleting `.`, so that `find -delete` (or `find . -delete`) doesn't fail.

**Comment by sylvestre** [comment]
@Arcterus do you still want to land this? thanks

**Comment by dahc** [comment]
@sylvestre If @Arcterus doesn't, I do.

What's blocking it at this point? Just missing tests? I'd be happy to put some together if needed (separate PR, I guess)

**Comment by sylvestre** [comment]
@dahc Yeah, rebase (all files moved) + rustfmt + probably some clippy fixes + tests
(new PR is fine)

**Comment by dahc** [comment]
@sylvestre new PR: #82

**Pull request closed without merging**